### PR TITLE
PAT-951: add soc referal button and link

### PR DESCRIPTION
**Pull request by connormaglynn**
- add SOC referal button and link
- add feature toggle to disable this feature

### CODE REVIEW

File: helm_deploy/values-preprod.yaml [hunk 1]
```diff
@@ -40,6 +40,8 @@ env:
   PATHFINDER_UI_URL: https://preprod.pathfinder.service.justice.gov.uk/
   PATHFINDER_ENDPOINT_API_URL: https://preprod-api.pathfinder.service.justice.gov.uk/
   REDIS_ENABLED: true
+  SOC_URL: https://manage-soc-cases-preprodservice.justice.gov.uk
+  SOC_API_ENABLED: false
```
Comment (timharrison-moj) [inline helm_deploy/values-preprod.yaml:44]: No dot between preprod and service. Should be manages-soc-cases-preprod.service.justice.gov.uk for now. Though it will probably change domain name to reside under hmpps.service.justice.gov.uk soon.

Comment (connormaglynn) [reply helm_deploy/values-preprod.yaml:44]: Sorted :)

**Comment by jsrobertson** [comment]
Looks good to me. 👍

File: README.md [hunk 2]
```diff
@@ -116,3 +116,9 @@ now display 'run' icons for the class and each of its tests methods.
 - Gradle: https://gradle.org/
 - WireMock: http://wiremock.org/
 - Chromedriver: https://sites.google.com/a/chromium.org/chromedriver
+
+## Feature toggles
+- **SOC API:**
+This will enable/dislable to abilty to refer a prisoner to the SOC service and view their SOC profile if already referred.
```
Comment (petergphillips) [inline README.md:122]: typos on `disable` and `ability`

File: backend/config.js [hunk 3]
```diff
@@ -84,12 +84,18 @@ module.exports = {
       ui_url: process.env.PATHFINDER_UI_URL,
       timeoutSeconds: process.env.API_ENDPOINT_TIMEOUT_SECONDS || 30,
     },
+    soc: {
+      url: process.env.SOC_URL || '',
+      timeoutSeconds: process.env.API_ENDPOINT_TIMEOUT_SECONDS || 30,
```
Comment (petergphillips) [inline backend/config.js:89]: would be better to have specific timeout overrides per endpoint, original `API_ENDPOINT` is prison api which is slow

File: backend/services/prisonerProfileService.js [hunk 4]
```diff
@@ -5,14 +5,26 @@ const {
...
+  elite2Api,
+  keyworkerApi,
+  oauthApi,
+  dataComplianceApi,
+  pathfinderApi,
+  socApi,
+  systemOauthClient,
+}) => {
   const getPrisonerProfileData = async (context, offenderNo, username) => {
+    logger.info(`SOC API enabled: ${socEnabled}, SOC URl: ${socUrl}`)
```
Comment (petergphillips) [inline backend/services/prisonerProfileService.js:26]: this will log the SOC information every time someone goes to prisoner profile, will will fill up the logs

File: backend/services/prisonerProfileService.js [hunk 5]
```diff
@@ -104,6 +127,10 @@ module.exports = ({ elite2Api, keyworkerApi, oauthApi, dataComplianceApi, pathfi
         pathfinderUrl && pathfinderDetails && `${pathfinderUrl}nominal/${String(pathfinderDetails.id)}`,
       showPathfinderReferButton: Boolean(!pathfinderDetails && isPathfinderUser),
       pathfinderReferUrl: pathfinderUrl && `${pathfinderUrl}refer/offender/${offenderNo}`,
+      canViewSocLink: socEnabled && canViewSocLink,
+      socProfileUrl: socEnabled && socUrl && socDetails && `${socUrl}nominal/${String(socDetails.id)}`,
```
Comment (petergphillips) [inline backend/services/prisonerProfileService.js:131]: we were trying on all our new endpoints to have the `/` in the URL instead.  That way the URLs read better so would read as `${socUrl}/nominal` instead

File: feature.env [hunk 6]
```diff
@@ -13,4 +13,6 @@ TOKENVERIFICATION_API_URL=http://localhost:18100
 ALLOCATION_MANAGER_ENDPOINT_URL=http://localhost:18084/
 TOKENVERIFICATION_API_ENABLED=true
 OFFENDER_SEARCH_API_URL=http://localhost:18101
+OFFENDER_SEARCH_API_ENABLED=true
```
Comment (petergphillips) [inline feature.env:16]: you've enabled offender search by accident

File: helm_deploy/prison-staff-hub/templates/_envs.tpl [hunk 7]
```diff
@@ -169,5 +169,11 @@ env:
 
   - name: PATHFINDER_ENDPOINT_API_URL
     value: {{ .Values.env.PATHFINDER_ENDPOINT_API_URL | quote }}
+
+  - name: SOC_URL
```
Comment (petergphillips) [inline helm_deploy/prison-staff-hub/templates/_envs.tpl:173]: env variables look confused here - you've got `SOC_URL`, should it be `SOC_API_URL`?  Is this an `API` url or a front end URL?

Comment (connormaglynn) [reply helm_deploy/prison-staff-hub/templates/_envs.tpl:173]: SOC_URL is both API and front-end :)

File: backend/config.js (hunk 3 above)
Comment (petergphillips) [reply backend/config.js:89]: for example we default the offender search api below to 10 seconds instead since it is calling elastic search instead

File: feature.env (hunk 6 above)
Comment (connormaglynn) [reply feature.env:16]: Good spot there! Cheers Pete :)

Comment (connormaglynn) [reply feature.env:16]: Seems like it needs the true value to run the tests - but gotten rid of the duplicate false value at the bottom 👍

Comment (petergphillips) [reply feature.env:16]: no it needs the `false` value :confused:

Comment (connormaglynn) [reply feature.env:16]: Ahh may have been failing because there was a duplicate then, works with the false value :)

Comment (petergphillips) [reply feature.env:16]: so please remove the duplicate

File: backend/services/prisonerProfileService.js [hunk 8]
```diff
@@ -5,13 +5,23 @@ const {
   apis: {
     categorisation: { ui_url: categorisationUrl },
     pathfinder: { ui_url: pathfinderUrl },
+    soc: { url: socUrl, enabled: socEnabled },
     useOfForce: { prisons: useOfForcePrisons, ui_url: useOfForceUrl },
   },
   app: { notmEndpointUrl, displayRetentionLink },
 } = require('../config')
 const logErrorAndContinue = require('../shared/logErrorAndContinue')
+const logger = require('../log.js')
```
Comment (petergphillips) [inline backend/services/prisonerProfileService.js:14]: how did this pass linting?  Looks like `logger` isn't used :confused:
